mailpoet/lib/Form/DisplayFormInWPContent.php

```diff
@@ -123,8 +123,8 @@ private function shouldDisplay(): bool {
   private function displayFormInProductListPage(): bool {
     $displayCheck = $this->wp->applyFilters('mailpoet_display_form_in_product_listing', true);
 
-    $shopPageId = $this->woocommerceHelper->wcGetPageId('shop');
-    $this->wooShopPageId = $shopPageId && $shopPageId >= 0 ? $shopPageId : null;
+    $shopPageId = (int)$this->woocommerceHelper->wcGetPageId('shop');
+    $this->wooShopPageId = $shopPageId && $shopPageId > 0 ? $shopPageId : null;
```

Comment:
I wasn't sure why we were allowing values of `0` previously. I don't think WordPress posts can have post IDs of `0`, so I made this slight change just to ensure there's no possibility of setting `wooShopPageId` to `0`. Please let me know if I shouldn't have touched this logic.

Reply:
that's okay